### PR TITLE
Adding new job runner for split names based on dataset info cached response

### PR DESCRIPTION
**Pull request by AndreaFrancis**
As part of https://github.com/huggingface/datasets-server/issues/755 we will need a new job runner to compute split names from dataset-info response.

### CODE REVIEW

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          If the the previous step gave an error.
+        - [`~job_runners.split_names_from_dataset_info.PreviousStepFormatError`]
+            If the content of the previous step has not the expected format
+        - [`~libcommon.dataset.DatasetNotFoundError`]
+            If previous step content was not found for the dataset
+    </Tip>
+    """
+    logging.info(f"get split names from dataset info for dataset={dataset}, config={config}")
+    try:
+        response = get_response(kind="/parquet-and-dataset-info", dataset=dataset)
```
Comment (severo) [inline services/worker/src/worker/job_runners/split_names_from_dataset_info.py:91]:
```suggestion
        response = get_response(kind="/dataset-info", dataset=dataset)
```

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        raise DatasetNotFoundError("No response found in previous step for this dataset.", e) from e
+    if response["http_status"] != HTTPStatus.OK:
+        raise PreviousStepStatusError(
+            f"Previous step gave an error: {response['http_status']}. This job should not have been created."
+        )
+    content = response["content"]
+
+    if "dataset_info" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    dataset_info = content["dataset_info"]
```
Comment (severo) [inline services/worker/src/worker/job_runners/split_names_from_dataset_info.py:102]: these four lines should be removed if we depend on /dataset-info

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    if config not in dataset_info:
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    config_info = dataset_info[config]
```
Comment (severo) [inline services/worker/src/worker/job_runners/split_names_from_dataset_info.py:106]:
```suggestion
    config_info = content[config]
```

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        raise PreviousStepStatusError(
+            f"Previous step gave an error: {response['http_status']}. This job should not have been created."
+        )
+    content = response["content"]
+
+    if "dataset_info" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    dataset_info = content["dataset_info"]
+
+    if config not in dataset_info:
```
Comment (severo) [inline services/worker/src/worker/job_runners/split_names_from_dataset_info.py:104]:
```suggestion
    if config not in content:
```

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    config_info = dataset_info[config]
+
+    if "splits" not in config_info:
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    split_name_items: List[SplitNameItem] = [
+        {"dataset": dataset, "config": config, "split": str(split)} for split in config_info["splits"]
```
Comment (severo) [inline services/worker/src/worker/job_runners/split_names_from_dataset_info.py:111]: maybe put these lines in a try block, in order to raise PreviousStepFormatError if the format is not as expected (a dictionary, from which we take the keys)

**Review by severo — APPROVED** [review]
Some details

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py (hunk 2 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/split_names_from_dataset_info.py]: Done, thanks!

File: services/worker/src/worker/job_runners/split_names_from_dataset_info.py (hunk 5 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/split_names_from_dataset_info.py]: Done, thanks!

**Review by severo — APPROVED** [review]

**Pull request merged**